model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -275,12 +277,48 @@ class ModelReplicationServer(val repositoriesManager: RepositoriesManager) {
     }
 
     private suspend fun ApplicationCall.respondDelta(versionHash: String, baseVersionHash: String?) {
-        val delta = VersionDelta(
-            versionHash,
-            baseVersionHash,
-            objectsMap = repositoriesManager.computeDelta(versionHash, baseVersionHash),
-        )
-        delta.checkObjectHashes()
-        respond(delta)
+        when (val format = request.queryParameters["format"]) {
+            "versionHash" -> {
+                respondText(versionHash)
+            }
+            "plain" -> {
+                respondTextWriter {
+                    append(versionHash)
+                    var first = true
+                    repositoriesManager.computeDelta(versionHash, baseVersionHash).collect {
+                        HashUtil.checkObjectHash(it.first, it.second)
+                        if (first) {
+                            first = false
+                            check(it.first == versionHash) { "First object should be the version" }
+                        } else {
+                            appendLine()
+                            append(it.first)
+                        }
+                        appendLine()
+                        append(it.second)
+                    }
+                }
```

Comment:
I'd factor this into a separate method to avoid this endpoint from growing.

Reply:
It's already a separate method.

Reply:
I see all three formats being implemented in the same route function in ktor?

Reply:
I refactored it a bit, but the method is now just 30 lines. Splitting it wouldn't improve the readability.